### PR TITLE
Use a Positron-specific name for Linux icon

### PR DESCRIPTION
**Pull request by juliasilge**
Addresses #4875 

This string is used in https://github.com/posit-dev/positron/blob/91be0db898da1afca6873445b6c402806a77d881/build/gulpfile.vscode.linux.js when the Linux build is created, and we need to have a filename that doesn't conflict with VS Code.

### QA Notes

- Create a Linux release build
- `ar -x Positron-2099.12.0-999.deb`
- `tar -xvf data.tar.xz`
- Look at the filename located at `usr/share/pixmaps/`

### CODE REVIEW

File: product.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
 	"win32TunnelServiceMutex": "positron-tunnelservice",
 	"win32TunnelMutex": "positrontunnel",
 	"darwinBundleIdentifier": "com.rstudio.positron",
-	"linuxIconName": "com.visualstudio.code.oss",
+	"linuxIconName": "com.rstudio.positron",
```
Comment (petetronic) [inline product.json:33]: Any thoughts on `co.posit.positron`?

Comment (juliasilge) [reply product.json:33]: Sure! Do you think I should also change `darwinBundleIdentifier`

**Comment by juliasilge** [comment]
I [made a build here](https://github.com/posit-dev/positron-builds/actions/runs/11484969899) where we can confirm this works, but I'll wait to get confirmation on how many keys in `product.json` to change.

**Comment by sharon-wang** [comment]
I downloaded the DEB build from the provided link and followed the QA notes above.

On Ubuntu 24, I'm seeing:

```sh
usr/share/pixmaps$ ls
co.posit.positron.png
```

**Review by sharon-wang — APPROVED** [review]

**Pull request merged**